Add simple site creation with blueprint

zaerl opened the pull request:
This PR adds a blueprint loader for both newly created and preexisting free sites.

See: https://linear.app/a8c/issue/PLAYGRD-239/create-golden-path-to-create-a-simple-site-with-a-predefined-blueprint

## Proposed Changes

* Remove Business/WooCommerce filter.
* Remove redirect to import page for playground free sites
* Add `blueprint` field to site creation endpoint
* Add check on current running `rest/v1.1/sites/__ID__imports/`
* Add support to `importerWordpress` page

## Why are these changes being made?
<!--
It's easy to see what a PR does but much harder to find out why it was made,
particularly when researching old changes in history. Record an explanation of
the motivation behind this change and how it will help.
-->

* To let the user use blueprints in all kinds of plans
* To let the user apply a blueprint

## Testing Instructions

> [!TIP]
> If your user has a security key enabled, you will not log in to http://calypso.localhost:3000, because it's not https. What you can do is to change the `AUTH` field in the bottom right DEV menu to to `OAUTH`. This will automatically log in you. If this not work be sure to use another user without a key.

Create a site from a blueprint:

* Apply 198037-ghe-Automattic/wpcom and follow testing instructions
* Checkout `add/new-site-from-blueprint`
* `yarn start`
* Select a free site
* Visit `http://calypso.localhost:3000/setup/site-setup/importerWordpress?siteSlug=__BLOG_ID__.wordpress.com`
* Select a blueprint JSON
* The import page refreshes every five seconds. You should see some confetti


https://github.com/user-attachments/assets/1bb6084d-65ae-4147-aff6-a80b2fb3d386

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you tested accessibility for your changes? Ensure the feature remains usable with various user agents (e.g., browsers), interfaces (e.g., keyboard navigation), and assistive technologies (e.g., screen readers) (PCYsg-S3g-p2).
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

zaerl committed:
Add simple site creation with blueprint

zaerl committed:
Add simple site creation with blueprint

zaerl committed:
Add simple site creation with blueprint

zaerl committed:
Add simple site creation with blueprint

zaerl committed:
Add support for free plan redirect

zaerl committed:
Add support for free plan redirect

zaerl committed:
Add support for free plan redirect

chriskmnds commented on client/landing/stepper/declarative-flow/flows/onboarding/onboarding.ts:
```diff
@@ -1,5 +1,6 @@
 import { isEnabled } from '@automattic/calypso-config';
 import { OnboardActions, OnboardSelect } from '@automattic/data-stores';
+import { FREE_PLAN_PRODUCT_ID } from '@automattic/data-stores/src/plans';
```
I think the idea is to gradually stop using the `data-stores` package - although I'm not sure if that's only relevant to the v2 dashboard, or in general. I'm also not sure where something like this would fit - in the dashboard's product `utils` folder? I'm honestly a bit puzzled/confused about how we intend to normalize and centralize all the data (types, constants, selectors). So a simple question for arch folks (everyone) - where would a `FREE_PLAN_PRODUCT_ID` live that can be potentially consumed within and outside the v2 dashboard? (@Automattic/lego ?)

That said, there is also a selector `isPlanProductFree`, which might be better than importing the ID and conditioning inline. 🤷

fushar replied:
I think we as an Architecture group need a bigger discussion on how to share more code between onboarding and dashboard. We managed to share APIs and queries via `api-core` and `api-queries` packages but that's it. Plans and products are still not yet solved.

I'd recommend just using an existing selector (`isPlanProductFree`) for now.

chriskmnds replied:
> I think we as an Architecture group need a bigger discussion on how to share more code between onboarding and dashboard. We managed to share APIs and queries via `api-core` and `api-queries` packages but that's it. Plans and products are still not yet solved.
> 
> I'd recommend just using an existing selector (`isPlanProductFree`) for now.

There's all the stuff under `utils`, which has relevant helpers/selector-like functions e.g. [purchase.ts ](https://github.com/Automattic/wp-calypso/blob/trunk/client/dashboard/utils/purchase.ts) has things like `isDIFMProduct`, `isTieredVolumeSpaceAddon`, `isSiteRedirect`, etc.

These don't feel all that discoverable to me.  `purchase.ts` may be helping with the purchases UI, but they are technically generalised product-level semantics - just some accepting a `purchase` parameter, others a `product`, yet all working over a `product_slug`. These need both cleanup and extraction to a more common location (make it discoverable). I think this is what `calypso-products` aimed to serve, but I understand we no longer want to refer to that. The need for a common location will surface sooner or later IMO (and not sure why it needs to "resurface" - we should already know the need for it). :-)

> The need for a common location will surface sooner or later IMO (and not sure why it needs to "resurface" - we should already know the need for it)

Sorry - not judgment! :-)

zaerl replied:
Thank you very much, folks. I didn't notice the plans selector. Fixed.

zaerl committed:
Substitute manual check with isPlanProductFree() call

zaerl committed:
Substitute manual check with isPlanProductFree() call

zaerl committed:
Substitute manual check with isPlanProductFree() call

zaerl committed:
Substitute manual check with isPlanProductFree() call

zaerl committed:
Fix: remove additional check

zaerl committed:
Fix: remove additional check

zaerl committed:
Fix: remove additional check

bgrgicak reviewed (commented):
I've tried it a few times with my sandbox on `trunk` (your PR was merged) and async job watcher running, but I can't see the `playground_apply_blueprint` job run.
Note: I've pointed both wordpress.com and public-api.wordpress.com to my sandbox.

bgrgicak commented on client/landing/stepper/declarative-flow/flows/onboarding/onboarding.ts:
```diff
@@ -227,18 +242,20 @@ const onboarding: FlowV2< typeof initialize > = {
-										{
+							const redirectTo: string =
+								playgroundId &&
+								! isPlanProductFree( {} as unknown as State, planCartItem?.product_id )
+									? addQueryArgs( withLocale( '/setup/site-setup/importerPlayground', locale ), {
 											siteSlug,
-										}
-								  );
+											siteId: providedDependencies.siteId,
+											playground: playgroundId,
+									  } )
+									: addQueryArgs(
+											withLocale( '/setup/onboarding/post-checkout-onboarding', locale ),
+											{
+												siteSlug,
+											}
+									  );
```
Does this mean that we won't support imports of Playground sites for the free plan? (I'm not sure if we ever supported it)

I find the flow confusing as I expect to see the site I just created in Playground in the free plan and not the site that initially loaded from the Blueprint.

It might be worth to use the Blueprint Extractor plugin or similar to extract a Blueprint from the current site and use that to create a site.

zaerl replied:
> Does this mean that we won't support imports of Playground sites for the free plan? (I'm not sure if we ever supported it)

No, we do. We didn't before the latest changes. See https://github.com/Automattic/wp-calypso/pull/106718/files#diff-9e68e7a37dacf21caf0ed16ddc6c3259b444f6456b1448b4eeb4672d578bbe85R262

This code here skips the migration page for free plans; otherwise, you will find a page that asks you to buy a Business or WooCommerce plan.

gabrielcaires committed:
Add hamburger menu (#107742)

gabrielcaires committed:
Use same MSD strategy to evaluate if the user show see the new reader (#107749)

gabrielcaires committed:
Use helps to point to the correct dashboard URL (#107750)

wellyshen committed:
AI-776: Integrating Odie conversation list into the conversation history view (#107608)

* AI-776: Adding Odie and Zendesk conversations to the conversation history view

* Improve convo list hooks names

* WIP: Integrating Odie convos and code refactors

* Fix: `formatConversationDate` logic

* Implement more stuff

* Improve components

* Add TODO

* Add TODO

* Update comment

* Install `@wordpress/url` package

* Add comments

* Format code

* Display convo type

* Make Odie convo work

* Export `isLoading` from `useManagedOdieChat`

* Improve `useConversationList` perf

* Revert `ConversationListItem` HTML tags

* Address GH review issues

* Update comment

* Update comment

* Update README.md

* Reuse modules, types, constant from odie-client instead

* Uninstall `@wordpress/url` package

* Fix: Empty view show up when loading Odie convos

* Code refactor

* use `odieInteractionId` instead

* Simplify code

* Fix: Avatar class names

* Address GH review issues

* Add comment

* Fix: Compare day by using the same local day

* Fix: Odie convo API data type

* Remove `type` from `getSupportInteractionId()`

* Use `addQueryArgs` to set URL params

* Reuse `useGetOdieConversations` from odie-client

* Fix: Sort convos by `message.received`

* Code refactor

* Valid Orc convos & Fix convo item display logic

* Update comment

* Use `generateConversationSubtitle()` to handle convo subtitle

* Adjust convo item aria-label

* Update header comment

* Add error handling for the convo item compo

* Reformat code

* Add error handling for `formatConversationDate()`

* Fix: Improve `generateConversationTitle()` logic

* Address GH review issues

* Revert fixing the loading state

* Shorten state names

* Remove debug convo item subtitle

* Clean Odie convo list relevant code

* Fix: Potential state overlapped issue

* Revert the aria-label of `<ConversationListItem>`

* Update comment

---------

Co-authored-by: Welly Shen <welly.shen@automattic.com>

xavier-lc committed:
Emails Dashboard: Reduce query count on the Add Email Forwarder form (#107747)

xavier-lc committed:
Emails Dashboard: Prevent same forwarder from being added twice (#107745)

Copons committed:
Theme Showcase: Retrieve the commercial info for Community themes (#107771)

* Bump the Dotorg API themes endpoint version to 1.2.
* Include the commercial info in the theme object used by the `ThemeSheet` component.

mehmoodak committed:
Reader: Fix notifications keyboard shortcuts popover on MSD header (#107773)

Updated CSS of the "Keyboard Shortcuts" popover.

mehmoodak committed:
Reader: Fix logout on reader MSD header (#107772)

- Extracted the logout functionality from auth to a separate function.
- Used that function in MSD header (new layout) and in the Reader header.

jcheringer committed:
Agents Manager on MSD: Fix load and adjust styles (#107695)

* Agents Manager on MSD: Fix load and adjust styles

* Fix test

michaeldcain committed:
MSD: add Store Sandbox helper environment badge (#107777)

* Add Store Sandbox helper to MSD environments

* Adjust styles

* Add enabled color

zaguiini committed:
MSD Bulk actions: Stop polling endpoint if there are no jobs (#107775)

heavyweight committed:
Fix jp test (#107780)

heavyweight committed:
Fix/dotcom 15382 (#107462)

* Apply design feedback

* Apply design feedback

leonardost committed:
Domain Transfer Redesign: Fix margin of the transfer setup instructions (#107662)

* Fix margin of the transfer setup instructions

* Pass `size` proeprty to CollapsibleCard

* Remove custom CSS for the transfer setup

* Set inline styling and custom size property in the Transfer Setup component

* Fix domain connection layout

* Remove unused CSS style file import

* Add `domain-transfer-redesign` feature flag to dashboard-* config files

xavier-lc committed:
Plugins Dashboard: Make `Delete` action available at all times (#107723)

paulopmt1 committed:
Add Survicate to the Calypso help center (#107757)

* Add Survicate to Calypso's help center

* Make sure we only show the feedback CTA when survicate is loaded

* Make sure the FF works as expected

* Fix types

* Enable the new feature on production but keep the gate to simplify its control just in case

* Add tests for the HelpCenterMoreResoruces section

p-jackson committed:
Show flash message in MSD site list after accepting a site invite (#107730)

leonardost committed:
Enable "survicate_enabled_at_help_center" feature flag (#107791)

fushar committed:
MSD: add busy state to action button while loading purchases in delete account (#107769)

chickee0311 committed:
Update states list for CCPA optout (#107677)

p-jackson committed:
Wrap people URLs with wpcomLink() (#107792)

taipeicoder committed:
Update site_name for Dashboard environments (#107794)

yashwin committed:
A4A > Marketplace: Implement term pricing toggle (#107706)

* A4A > Marketplace: Implement term pricing toggle

* address PR comments

p-jackson committed:
MSD: Update dashboard e2e tests to use new URL without breaking TeamCity jobs (#107760)

p-jackson committed:
Fix SiteSettingsPage.launchSite button which is a link, not a button (#107797)

travisw committed:
A4A: Never use selectedSite for cart key in A4A checkout (#107793)

heavyweight committed:
Onboarding: setup post checkout step (#107221)

* Onboarding: setup post checkout step

* Add no site and migration navigation

* Add ff and wire it to the flow

* Setup ai build with existing site

* minor cleanup

* Apply initial feedback

* wire ai eligibility check

* Apply design feedback

* Fix ident message

* Revert unnecessary changes

* Remove plan check

* fixup after merge, fix loader

* Remove unused step

* minor fixup

* Update loading

* return of the dot

chriskmnds committed:
Connect Refresh: Authorise/Connect screen i1 wireframe (#105309)

* Add UI for new OAuth

---------

Co-authored-by: Kateryna Kodonenko <kateryna@automattic.com>

katinthehatsite committed:
Ensure we pass autopush to Studio (#107779)

Co-authored-by: Kateryna Kodonenko <kateryna@automattic.com>

xavier-lc committed:
Plugins Dashboard: Clean leftover code (#107783)

hambai committed:
Fix the snackbar notice to show the server error message when setting primary domain via the dataview actions (#107737)

escapemanuele committed:
Help Center: Fix links in messages (#107770)

* Fix links in messages

* Maintain underline on hover

* Color on hover

escapemanuele committed:
4096 to 1000 (#107782)

gius80 committed:
Fix domain connection styles overridden by component CSS loading order in my.wordpress.com (#107765)

* Fix css specificity

* Add explanation for !important

* Fix action list padding

katinthehatsite committed:
Add sections (#107801)

Co-authored-by: Kateryna Kodonenko <kateryna@automattic.com>

fushar committed:
MSD: fix link for Delete domain button (#107796)

xavier-lc committed:
Plugins Dashboard: Get rid of enigmatic `onSuccess` function (#107786)

xavier-lc committed:
Plugins Dashboard: Show plugin name even if user doesn't have it installed on any site (#107784)

hambai committed:
Add MSD Domain Diagnostics screen and overview Setting (#107758)

Add MSD Domain Diagnostics screen and overview Setting

gcsecsey committed:
Marketplace: show most recent plugin version for marketplace products (#107751)

* Marketplace: show plugin verson for marketplace products

* use the plugin version on plugin management page

* make ssr and hydrated versions consistent

xavier-lc committed:
MSD Plugins: Make `PluginSwitcher` look more like `DataViews` (#107805)

arthur791004 committed:
MSD Domains: Preload missing suspense query on domain overview (#107795)

* MSD Domains: Preload missing suspesnse query on domain overview

* Address feedback

fushar committed:
MSD: use DataViews loading state for Active Upgrades (#107800)

heavyweight committed:
HC: ensure user messages are parsed (#107803)

* HC: ensure user messages are parsed

* extend check

zaerl committed:
Minor doc change

zaerl committed:
Minor doc change

zaerl committed:
Minor doc change

zaerl committed:
Update onboarding.ts

sirbrillig committed:
MSD: Remove "downgrade" language from PurchaseSettings (#107811)

katinthehatsite committed:
OAuth: Improve design of the loading screen (#107743)

* Adjust loading screen

* Ensure content does not jump

---------

Co-authored-by: Kateryna Kodonenko <kateryna@automattic.com>

leonardost committed:
Domain Transfer Redesign: Redirect user to transfer setup page after checkout when they buy a single domain transfer (#107676)

* Redirect user to transfer setup page after checkout when they buy a single domain transfer

* Remove TODO comment

gabrielcaires committed:
Reader: Show app title on mobile + search menu item + layout adjustments (#107781)

* Show app title on mobile

* Spacing adjustments

* Enable feature flag on dev envs

* Add to remaning pages

* Add search bar title

* Refactor mobile header

* Fix quick-post padding

* Use js instead of css to hide the item

* Fix spacing

* Fix value

* Remove the parent selector

* Remove reader/quick-post flag

escapemanuele committed:
Better check for live chat (#107809)

jcheringer committed:
Fix main content not scrollable when sidebar is open (#107788)

* Fix main content not scrollable when sidebar is open

* Address feedback

sirbrillig committed:
Remove "Plan features" from MSD PurchaseSettings (#107813)

yashwin committed:
A4A > Marketplace: Fix upsell checkout (#107768)

* A4A > Marketplace: Fix upsell checkout

* address PR comment

xavier-lc committed:
MSD Plugins: Add sites with updates filter (#107808)

StevenDufresne committed:
MSD: Link whois privacy card to privacy view. (#107649)

fushar committed:
MSD: remove /domains redirection which conflicts with local MSD /domains (#107810)

fushar committed:
MSD: update opt-in / opt-out survey link (#107819)

arthur791004 committed:
MSD: Introduce OptInWelcomeModal (#107799)

* MSD: Introduce OptInWelcomeModal

* Allow user to dismiss the modal by clicking outside

* Revert "Allow user to dismiss the modal by clicking outside"

This reverts commit 7a1d7ab7ad940fa720e3b69e29225e0242ae38ed.

* Update copy for consistency

arthur791004 committed:
MSD: Don't show opt-in welcome modal on backport (#107821)

taipeicoder committed:
Remove support for /v2 and /ciab routes in the Calypso environment (#107735)

fushar committed:
MSD: Launch the opt-in period for existing users (#107822)

arthur791004 committed:
MSD: Fix shouldIncludeA8COwned option (#107824)

* MSD: Fix shouldIncludeA8COwned option

* Apply changes to CIAB

* Fix Status filter

* Fix error

fredrikekelund committed:
Use static strings for VAT form in checkout (#107532)

* Use static strings for VAT form in checkout

* Fix tests

* Improve translator comments

arthur791004 committed:
Domain Flow: Fix the back button on the UseMyDomain screen (#107764)

zaerl committed:
Merge branch 'add/new-site-from-blueprint' of https://github.com/Automattic/wp-calypso into add/new-site-from-blueprint
